gax-java/gax/src/test/java/com/google/api/gax/util/TimeConversionUtilsTest.java

```diff
@@ -69,4 +69,42 @@ void testToThreetenTimeInstant_validInput_succeeds() {
         jtInstant.toEpochMilli(), TimeConversionUtils.toThreetenInstant(jtInstant).toEpochMilli());
     assertNull(TimeConversionUtils.toThreetenInstant(null));
   }
+
+  @Test
+  void testToThreeteenInstant_bigInput_doesNotOverflow() {
+    // defaults to MAX_SECONDS plus the max value of long for the nanos part
+    java.time.Instant jtInstant = java.time.Instant.MAX;
+    org.threeten.bp.Instant ttInstant = TimeConversionUtils.toThreetenInstant(jtInstant);
+    assertEquals(jtInstant.getEpochSecond(), ttInstant.getEpochSecond());
+    assertEquals(jtInstant.getNano(), ttInstant.getNano());
+  }
+
+  @Test
+  void testToJavaTimeInstant_bigInput_doesNotOverflow() {
+    // defaults to MAX_SECONDS plus the max value of long for the nanos part
+    org.threeten.bp.Instant ttInstant = org.threeten.bp.Instant.MAX;
+    java.time.Instant jtInstant = TimeConversionUtils.toJavaTimeInstant(ttInstant);
+    assertEquals(jtInstant.getEpochSecond(), ttInstant.getEpochSecond());
+    assertEquals(jtInstant.getNano(), ttInstant.getNano());
+  }
+
+  @Test
+  void testToThreeteenDuration_bigInput_doesNotOverflow() {
+    // we use the max long value for the seconds part and an arbitrary int for the nanos part, so we
+    // can confirm that both components are preserved
+    java.time.Duration jtDuration = java.time.Duration.ofSeconds(Long.MAX_VALUE, 123);
```

Comment:
qq, any specific reason for an arbitrary value of 123 and not Long/Integer.MAX_VALUE? Perhaps we use MAX_VALUE for consistency?

Reply:
No specific reason. I modified this to use the max possible nano value without overflowing the duration